tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenWithThread.cs

```diff
@@ -0,0 +1,59 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Misc;
+
+namespace MongoDB.Driver.Tests.JsonDrivenTests
+{
+    public class JsonDrivenWithThreadContext
+    {
```

Comment:
Nit: Should this context class be declared below the main class? The file is named `JsonDrivenWithThread.cs` so I expected the first class that I saw to be `JsonDrivenWithThread` and not `JsonDrivenWithThreadContext`

Reply:
>so I expected the first class

a reasonable assumption, but at the same time, I guess it makes sense to place this context at the base class for test steps that use "threads" because this context will be used in all of them. So, I don't insist on the current solution but I prefer it a bit more :)